src/mlpack/methods/fastmks/fastmks_main.cpp

```diff
@@ -131,6 +131,9 @@ void BINDING_FUNCTION(util::Params& params, util::Timers& timers)
         "no output will be saved");
   }
 
+  // input data can't be empty
+  RequireNonEmptyInputValue(params, "reference", true, " can not be empty!");
```

Comment:
Just to provide an example of what I mean, this particular binding has several input parameters, and there are even cases where it is *okay* when `reference` is empty (for instance, it might not be passed when `input_model` is passed).  There is also the `query` parameter, and that has some restrictions on its shape.  I hope this is helpful in explaining what I mean. :+1:

Reply:
yes, got it